src/components/nowPlayingBar/nowPlayingBar.js

```diff
@@ -146,13 +157,38 @@ define(['require', 'datetime', 'itemHelper', 'events', 'browser', 'imageLoader',
             }
         });
 
-        elem.querySelector('.previousTrackButton').addEventListener('click', function () {
+        elem.querySelector('.previousTrackButton').addEventListener('click', function (e) {
+            if (currentPlayer) {
+                if (currentPlayer.id === 'htmlaudioplayer' && (currentPlayer._currentTime >= 5 || !playbackManager.previousTrack(currentPlayer))) {
+                    // Cancel this event if doubleclick is fired
+                    if (e.detail > 1 && playbackManager.previousTrack(currentPlayer)) {
+                        return;
+                    }
+                    playbackManager.seekPercent(0, currentPlayer);
+                    // This is done automatically by playbackManager, however, setting this here gives instant visual feedback
+                    positionSlider.value = 0;
+                } else {
+                    playbackManager.previousTrack(currentPlayer);
+                }
+            }
+        });
 
+        elem.querySelector('.previousTrackButton').addEventListener('dblclick', function () {
             if (currentPlayer) {
                 playbackManager.previousTrack(currentPlayer);
             }
         });
 
+        elem.querySelector('.btnShuffle').addEventListener('click', function () {
+            if (currentPlayer) {
+                if (playbackManager.getPlaylistShuffleMode(currentPlayer) === 'Sorted') {
```

Comment:
@MrTimscampi FYI this was my initial implementation, in order to keep things simple :). Changed it to using values to keep consistency (regarding Repeat modes) and also because I thought that maybe external third party players that use our playbackManager or some crazy functionality we add in the future could simply keep shuffling without switching modes.

Didn't thought about re-adding it as a helper function and I think it's excellent because its simple enough and gives full control.

Is there any problem if I add it and simply leave it unused? Or shall I change all my references in remotecontrol and nowplayingbar to it?

Reply:
Ideally, nothing should be unused ;)

If it's not much work, feel free to add it and change references. Otherwise, simply adding a TODO will do for now :)